opcua/adapter.go

```diff
@@ -102,37 +103,67 @@ func (as *adapterService) RemoveChannel(ctx context.Context, chanID string) erro
 	return as.channelsRM.Remove(ctx, chanID)
 }
 
-func (as *adapterService) ConnectThing(ctx context.Context, chanID, thingID string) error {
+func (as *adapterService) ConnectThing(ctx context.Context, chanID string, thingIDs []string) error {
 	serverURI, err := as.channelsRM.Get(ctx, chanID)
 	if err != nil {
 		return err
 	}
 
-	nodeID, err := as.thingsRM.Get(ctx, thingID)
-	if err != nil {
-		return err
-	}
+	for _, thingID := range thingIDs {
+		nodeID, err := as.thingsRM.Get(ctx, thingID)
+		if err != nil {
+			return err
+		}
 
-	as.cfg.NodeID = nodeID
-	as.cfg.ServerURI = serverURI
+		as.cfg.NodeID = nodeID
+		as.cfg.ServerURI = serverURI
 
-	c := fmt.Sprintf("%s:%s", chanID, thingID)
-	if err := as.connectRM.Save(ctx, c, c); err != nil {
-		return err
-	}
+		c := fmt.Sprintf("%s:%s", chanID, thingID)
+		if err := as.connectRM.Save(ctx, c, c); err != nil {
+			return err
+		}
 
-	go func() {
-		if err := as.subscriber.Subscribe(ctx, as.cfg); err != nil {
-			as.logger.Warn(fmt.Sprintf("subscription failed: %s", err))
+		go func() {
+			if err := as.subscriber.Subscribe(ctx, as.cfg); err != nil {
+				as.logger.Warn("subscription failed", slog.Any("error", err))
+			}
+		}()
+
+		// Store subscription details
+		if err := db.Save(serverURI, nodeID); err != nil {
+			return err
 		}
-	}()
+	}
 
-	// Store subscription details
-	return db.Save(serverURI, nodeID)
+	return nil
 }
 
-func (as *adapterService) Browse(ctx context.Context, serverURI, namespace, identifier string) ([]BrowsedNode, error) {
-	nodeID := fmt.Sprintf("%s;%s", namespace, identifier)
+func (as *adapterService) Browse(ctx context.Context, serverURI, namespace, identifier, identifierType string) ([]BrowsedNode, error) {
+	var nodeID string
+
+	switch identifierType {
```

Comment:
You can simplify a little bit with something like:
```go
func (as *adapterService) Browse(ctx context.Context, serverURI, namespace, identifier, identifierType string) ([]BrowsedNode, error) {
	idFormat := "s"
	switch identifierType {
	case "string": // Only for the sake of the efficiency, since the first case is the most likely to happen.
		break
	case "numeric":
		if _, err := strconv.Atoi(identifier); err != nil {
			args := []any{
				slog.String("namespace", namespace),
				slog.String("identifier", identifier),
				slog.Any("error", err),
			}
			as.logger.Warn("failed to parse numeric identifier", args...)
			break
		}
		idFormat = "i"
	case "guid":
		idFormat = "g"
	case "opaque":
		idFormat = "b"
	}
	nodeID := fmt.Sprintf("ns=%s;%s=%s", namespace, idFormat, identifier)
	nodes, err := as.browser.Browse(serverURI, nodeID)
	if err != nil {
		return nil, err
	}
	return nodes, nil
}
```